Sources/SpeziHealthKit/Sample Types/SampleType.swift

```diff
@@ -0,0 +1,199 @@
+    private let variant: Variant
+    
+    public var id: String {
+        hkSampleType.identifier
+    }
+    
+    /// Creates a new ``HealthKitSampleType``.
+    /// - Note: Don't use this initializer directly. Instead, use one of the static factory methods: `HealthKitSampleType.quantity`, `.correlation`, and `.category`.
+    private init(_ hkSampleType: Sample._SampleType, displayTitle: String, variant: Variant) {
+        self.hkSampleType = hkSampleType
+        self.displayTitle = displayTitle
+        self.variant = variant
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(id)
+    }
+    
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.id == rhs.id
+    }
+    
+    public static func == (lhs: Self, rhs: HealthKitSampleType<some Any>) -> Bool {
+        lhs.id == rhs.id
+    }
+}
+
+
+extension HealthKitSampleType where Sample == HKQuantitySample {
+    /// The recommended unit that should be used when displaying values of this sample type to a user.
+    public var displayUnit: HKUnit {
+        switch variant {
+        case .quantity(let displayUnit, _):
+            return displayUnit
+        case .correlation, .category, .electrocardiogram, .audiogram:
+            // SAFETY:
+            // This branch is unreachable; the initializers are defined and structured in a way that all
+            // `HealthKitSampleType<HKQuantitySample>` objects always must specify a displayUnit.
+            fatalError("Cannot provide '\(#function)' for '\(Self.self)'")
+        }
+    }
+    
+    public var expectedValuesRange: ClosedRange<Double>? {
+        switch variant {
+        case .quantity(displayUnit: _, let expectedValuesRange):
+            return expectedValuesRange
+        case .correlation, .category, .electrocardiogram, .audiogram:
+            // SAFETY:
+            // This branch is unreachable; the initializers are defined and structured in a way that all
+            // `HealthKitSampleType<HKQuantitySample>` objects always must specify an expectedValuesRange.
+            fatalError("Cannot provide '\(#function)' for '\(Self.self)'")
+        }
+    }
+}
+
+
+extension HealthKitSampleType where Sample == HKCorrelation {
+    /// The recommended unit that should be used when displaying values of this sample type to a user.
+    public var displayUnit: HKUnit? {
+        switch variant {
+        case .correlation(let displayUnit):
+            return displayUnit
+        case .quantity, .category, .electrocardiogram, .audiogram:
+            // SAFETY:
+            // This branch is unreachable; the initializers are defined and structured in a way that all
+            // `HealthKitSampleType<HKCorrelation>` objects always must specify a displayUnit.
+            fatalError("Cannot provide '\(#function)' for '\(Self.self)'")
+        }
+    }
+}
+
+
+// MARK: Factory methods for commonly-used sample types
+
+extension HealthKitSampleType {
```

Comment:
We might want to make these public if we don't expect to create individual static values for all quantity types (and even then it might be good to have an escape hatch if needed when new sample types are added).